### PR TITLE
Make /ai command

### PR DESCRIPTION
**Pull request by DeathlyPlays**
Credit goes to AeonLucid for sniping me to it

### CODE REVIEW

**Comment by DeathlyPlays** [comment]
We have tested it and we have proof it works

**Comment by sparkychildcharlie** [comment]
You need to escape HTML

**Comment by DeathlyPlays** [comment]
@Lord-Haji

**Comment by bumbadadabum** [comment]
I honestly don't see why this should be a command. This is only going to
get spammed.

**Comment by AustinXII** [comment]
Agree with bumba

File: chat-plugins/info.js [hunk 1]
```diff
@@ -1950,6 +1951,24 @@ exports.commands = {
...
 		"!htmlbox [message] - Shows everyone a message, parsing HTML code contained. Requires: ~ & #",
 	],
+	'!ai': true,
+	askai: 'ai',
+	'8ball': 'ai',
+	ai: function (target, room, user) {
+		if (!this.runBroadcast()) return;
+		target = target.replace(/^[^a-z0-9]+/i, "");
+		let output = `<strong>Question:</strong> ${target}<br />`;
+		http.get(("http://qmarkai.com/qmai.php?q=" + target), res => {
```
Comment (Zarel) [inline chat-plugins/info.js]: You need to urlencode `target`

File: chat-plugins/info.js [hunk 2]
```diff
@@ -1950,6 +1951,24 @@ exports.commands = {
...
+			let data = '';
+			res.on('data', chunk => {
+				data += chunk;
+			}).on('end', () => {
+				output += `<strong>Answer:</strong> ${data.replace(/[\r\n]/g, "")}`;
```
Comment (Zarel) [inline chat-plugins/info.js]: Why are you ignoring linebreaks rather than forwarding them?

Comment (Zarel) [inline chat-plugins/info.js]: Also yeah, escape your stuff.

**Comment by Zarel** [comment]
Personally I think it'd be fun to have.

**Comment by DeathlyPlays** [comment]
We're gonna work on this and reopen later

**Comment by panpawn** [comment]
What about broadband usage on this?

**Comment by panpawn** [comment]
What's to stop someone from doing this command as fast as they can?  There should totally be some kind of throttle for it, that's stricter than the chatting throttle.

**Comment by DeathlyPlays** [comment]
@panpawn I suggest we make this a chat-commands.js thing and make it use the command cooldown thing that was made for broadcasting cooldown.

**Comment by sparkychildcharlie** [comment]
If this is a concern, it should be a throttle per IP - it's too easy to send say 500 bots spamming this at ~93 times a minute x 500 bots

**Comment by Zarel** [comment]
A global cooldown makes more sense, tbh.

**Comment by sparkychildcharlie** [comment]
I'm not sure if this (https://github.com/Zarel/Pokemon-Showdown/blob/master/chat-plugins/tcgtabletop.js) is the only place, but if a filter is used, maybe it should account for the other commadns that make HTTP requests as well

**Comment by DeathlyPlays** [comment]
Very true @sparkychildcharlie thanks for pointing that out, I didn't even consider that as a possibility.

**Pull request closed without merging**